src/main/java/Duke.java

```diff
@@ -1,10 +1,137 @@
+/**
+ * Solution below adapted from
+ * https://github.com/nus-cs2113-AY2021S1/contacts/blob/master/src/main/java/Contacts1.java
+ */
+
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final int MAX_NUMBER_OF_TASKS = 100;
+    private static Task[] tasks = new Task[MAX_NUMBER_OF_TASKS];
+    private static int taskNumber = 0;
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+        while (true) {
+            String command = getUserCommand();
+            processUserCommand(command);
+        }
```

Comment:
Good Job! 👍

I like how you abstracted your entire code into functions, allowing for each section to be readable and nameable without need for much comments.